nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -10,6 +10,8 @@ let
   toStr = value:
     if true == value then "yes"
     else if false == value then "no"
+    # escape according to https://www.php.net/manual/en/function.parse-ini-file.php
+    else if isString value then ''"${escape [ "\"" "\\" "$" ] value}"''
```

Comment:
Escaping `$` does not work, though – PHP will interpolate it even though you quoted it: https://github.com/NixOS/nixpkgs/issues/79469#issuecomment-631461513

Reply:
The quotes on the other hand will get stuck when using your patch:

```
test1=
test2="$2y$10$test2"
test3="\$2y\$10\$test3"
test4='$2y$10$test4'
test5='\$2y\$10\$test5'
test6=$2y$10$test6
```

Reply:
Very strange. Though I believe the escape does prevent `${x}` from being interpolated at least, which can appear anywhere in a string?

Not sure how to handle that case though, there are unfortunately some sequences that can't seem to be escaped... Another example that fails is newlines, despite the php.ini documentation implying that they're valid as-is inside quotes (see the "value concatenation" part for an example)...